src/utils/auth_utils.js

```diff
@@ -22,17 +30,33 @@ const firebaseConfig = {
 const app = initializeApp(firebaseConfig);
 const auth = getAuth(app);
 
+/**
+ * Signs a user in with Google using Firebase
+ * @returns A boolean indicating whether or not the user is new
+ */
 const signInWithGoogle = async () => {
-  // TODO
+  try {
+    const provider = new GoogleAuthProvider();
+    const userCredential = await signInWithPopup(auth, provider);
+
+    return getAdditionalUserInfo(userCredential).isNewUser;
+  } catch (err) {
+    return err;
+  }
 };
 
+/**
+ *
+ * @param {string} email The email to log in with
+ * @param {string} password The password to log in with
+ * @returns A boolean indicating whether or not the log in was successful
+ */
 const logInWithEmailAndPassword = async (email, password) => {
   try {
     await signInWithEmailAndPassword(auth, email, password);
-    alert('Sign in success');
+    return true;
   } catch (err) {
-    console.error(err);
-    alert(err.message);
+    return false;
```

Comment:
replace with throw, remove the `return true`. if nothing is thrown, we assume the method call was successful.